### PR TITLE
Update point cloud struct, Add Camera and Lidar stream in realtime

### PR DESCRIPTION
**Pull request by induZen05**
This PR has : 

- Updated Point Cloud Struct to save algo_type and motion_class
- Camera and Lidar support for realtime streaming

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -40,6 +40,20 @@ Source your .bash_profile (e.g. ~/.bashrc):
...
+
+```
+mkdir -p ~/external_driver_ws/src
+cd external_driver_ws/src
+git clone git@github.com:ZendarInc/spinnaker_sdk_camera_driver
+git clone git@github.com:ZendarInc/HesaiLidar_General_ROS
+cd ~/external_driver_ws
+catkin_make
+source ~/external_driver_ws/devel/setup.bash
+# add this to ~/.bashrc to make this permanent
```
Comment (MarcBrau) [inline README.md:53]: Would you mind making "# add this to ~/.bashrc to make this permanent" a bit clearer? I'd even go the way and write something like:
```
# Add the following line to your ~/.bashrc
source ~/external_driver_ws/devel/setup.bash
```

File: launch/zendar_camera.launch [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  <arg name="target_url" default="192.168.1.9"/>
+  <arg name="max_range" default="40"/>
+  <node pkg="zendar_ros_driver" type="zendar_driver_node" name="zendar_driver_node">
+    <param name="url" type="string" value="$(arg target_url)"/>
+    <param name="max_range" type="double" value="$(arg max_range)"/>
+  </node>
+    <!--START CAMERA NODE-->
+  <arg name="enable_camera" default="true"/>
+  <group if="$(arg enable_camera)">
+    <!--remap original spinnaker topics to the same as offline ros converter-->
```
Comment (MarcBrau) [inline launch/zendar_camera.launch:14]: I think this comment should be updated to reflect the usage in the external driver.

File: launch/zendar_full_system.launch [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  <arg name="enable_lidar" default="true"/>
+  <group if="$(arg enable_lidar)">
+    <!--remap original hesai topics to the same as offline ros converter-->
+    <remap from="/hesai/pandar" to="/zendar_driver_node/lidar_pointcloud"/>
+    <include file="$(find hesai_lidar)/launch/nyquist_lidar.launch"/>
+  </group>
+  <!--START CAMERA NODE-->
+  <arg name="enable_camera" default="true"/>
+  <group if="$(arg enable_camera)">
+    <!--remap original spinnaker topics to the same as offline ros converter-->
```
Comment (MarcBrau) [inline launch/zendar_full_system.launch:22]: Same comment as in `zendar_camera.launch` + two redundant white spaces in front of comment (sorry for being so nitpicky :smile: )

File: launch/zendar_camera.launch [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  <!--ROSBRIDGE SERVER FOR WEBVIZ-->
+  <include file="$(find rosbridge_server)/launch/rosbridge_websocket.launch"/>
+  <!--START SAR AND POINTCLOUD-->
+  <arg name="target_url" default="192.168.1.9"/>
+  <arg name="max_range" default="40"/>
+  <node pkg="zendar_ros_driver" type="zendar_driver_node" name="zendar_driver_node">
+    <param name="url" type="string" value="$(arg target_url)"/>
+    <param name="max_range" type="double" value="$(arg max_range)"/>
+  </node>
+    <!--START CAMERA NODE-->
```
Comment (MarcBrau) [inline launch/zendar_camera.launch:11]: Nitpick: Two stray whitespaces in front of comment.

File: launch/zendar_full_system.launch [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  <node pkg="tf2_ros" type="static_transform_publisher" name="lidar_roof_center_to_vehicle_tf"
+      args="0.0158 -0.03 0.026 0.013619330710412113 0.00929179537867042 0.6983536407713653 0.7155629733231164 vehicle lidar_roof_center"/> 
+  <node pkg="zendar_ros_driver" type="zendar_driver_node" name="zendar_driver_node">
+    <param name="url" type="string" value="$(arg target_url)"/>
+    <param name="max_range" type="double" value="$(arg max_range)"/>
+  </node>
+  <!-- START LIDAR NODE -->
+  <arg name="enable_lidar" default="true"/>
+  <group if="$(arg enable_lidar)">
+    <!--remap original hesai topics to the same as offline ros converter-->
```
Comment (MarcBrau) [inline launch/zendar_full_system.launch:15]: Same comment about external ros driver instead of offline ros converter.

**Review by MarcBrau — COMMENTED** [review]
Overall looks good to me. Just left some minor comments. Please excuse my nitpicking.

**Comment by Emmaquu** [comment]
Looks good to me. But I am not familiar how ros1 launch files work. I don't think we use launch files in ROS2.

**Review by Emmaquu — APPROVED** [review]

File: launch/zendar_camera.launch [hunk 6]
```diff
@@ -0,0 +1,18 @@
+<launch>
+  <!--ROSBRIDGE SERVER FOR WEBVIZ-->
+  <include file="$(find rosbridge_server)/launch/rosbridge_websocket.launch"/>
+  <!--START SAR AND POINTCLOUD-->
```
Comment (MarcBrau) [inline launch/zendar_camera.launch:4]: One more small thing, sorry I missed this yesterday. Do you mind removing the reference to SAR from this comment?

File: launch/zendar_full_system.launch [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+  <arg name="enable_lidar" default="true"/>
+  <group if="$(arg enable_lidar)">
+  <!--remap original hesai topics to the same as online ros converter-->
+    <remap from="/hesai/pandar" to="/zendar_driver_node/lidar_pointcloud"/>
+    <include file="$(find hesai_lidar)/launch/nyquist_lidar.launch"/>
+  </group>
+  <!--START CAMERA NODE-->
+  <arg name="enable_camera" default="true"/>
+  <group if="$(arg enable_camera)">
+  <!--remap original spinnaker topics to the same as offline ros converter-->
```
Comment (MarcBrau) [inline launch/zendar_full_system.launch:22]: "offline"

**Review by MarcBrau — APPROVED** [review]
LGTM. Thank you for addressing my comments, I left only two very minor things that slipped my first round. Sorry about that. I don't think that I have to review it again. Thank you for putting this together, it makes Zendar's ROS driver way more useful.

**Pull request merged**